AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/SQLite/StorageEngineAdapter+SQLite.swift

```diff
@@ -144,8 +144,15 @@ final class SQLiteStorageEngineAdapter: StorageEngineAdapter {
 
     func delete<M: Model>(_ modelType: M.Type,
                           withId id: Model.Identifier,
-                          completion: (DataStoreResult<Void>) -> Void) {
-        delete(untypedModelType: modelType, withId: id, completion: completion)
+                          completion: (DataStoreResult<M?>) -> Void) {
+        delete(untypedModelType: modelType, withId: id) { result in
```

Comment:
Hmm why are we changing this?

Reply:
the delete is no longer just a call to delete on storage but queries for the data and then deletes it. we pass the queried data back to the caller like we do with the `delete(modelType, predicate) -> [M]`  to send to the subscribers. we also take the queried data and store it in the MutationEvent for conflict resolution